src/DotNetWorker.Core/Context/DefaultFunctionContext.cs

```diff
@@ -58,6 +59,9 @@ public override IServiceProvider InstanceServices
 
         public override RetryContext RetryContext => Features.GetRequired<IExecutionRetryFeature>().Context;
 
+        // public override CancellationToken CancellationToken => base.CancellationToken;
+        public override CancellationToken CancellationToken => Features.GetRequired<IExecutionCancellationFeature>().CancellationToken;
```

Comment:
@fabiocav mentioned that we should not need to use a `Feature` here,  but without the Feature, the cancellation token was not getting cancelled / updated in the context from within the user app. In other words, the cancellation was not getting signaled and we never hit the `OperationCanceledException` in the sample. Looking for advice on the right way to do this.

Reply:
Was able to figure his out and removed the cancellation feature